## Some Sample Programs

#### 1.Write a program to find the factorial of the given number x.

In [1]:
def fact(x):
    if x == 0:
        return 0
    elif x == 1:
        return 1
    else:
        return x * fact(x-1)

fact(5)

120

#### 2.Write a program to find the factor of the given number

In [2]:
def factor(x):
    for i in range(1,x+1):
        if x % i == 0:
            print(i)
num = int(input("Enter the Number:"))
factor(num)

Enter the Number:10
1
2
5
10


#### 3.Write a program to find the given number is Armstrong number or not.

In [3]:
def armstrong(x):
    a = x
    sum = 0
    while x > 0:
        rem = x % 10 # Seperate the number
        sum = sum + (rem*rem*rem) # Cube and add
        x = x // 10 # Minimize the number
    
    if a == sum:
        print("Armstrong Number")
    else:
        print("Not an Armstrong Number")
        
num = int(input("Enter the Number:"))
armstrong(num)        

Enter the Number:153
Armstrong Number


#### 4. Write a program to print the Armstrong numbers between 100 to 999.

In [4]:
for i in range(100,1000):
    a = i
    sum = 0
    while i > 0:
        rem = i % 10 # Seperate the number
        sum = sum + rem**3 # Cube and add
        i = i // 10 # Minimize the number
    if sum == a:
        print(a)

153
370
371
407


#### 5. Write a program to count and print the number of odd and even numbers.

In [5]:
limit = int(input("Enter the Limit:"))

odd = []
even = []

for i in range(0,limit):
    num = int(input("Enter the Number:"))
    
    if num % 2 == 0:
        even.append(num)
    else:
        odd.append(num)
        
print(f"The Odd numbers are {odd} and the even numbers are {even}")
print(f"The Odd numbers counts: {len(odd)} and the even numbers counts: {len(even)}")

Enter the Limit:3
Enter the Number:23
Enter the Number:22
Enter the Number:12
The Odd numbers are [23] and the even numbers are [22, 12]
The Odd numbers counts: 1 and the even numbers counts: 2


#### 6.Write a program to find the reverse of n digit number using While loop

In [6]:
num = int(input("Enter the Number:"))
before = num
after = 0
while num > 0:
    rem = num % 10
    after = (after*10) + rem
    num = num // 10
    
print("Before:",before)
print("After:",after)

Enter the Number:1234
Before: 1234
After: 4321


#### 7.Write a program to find the Fibonacci Series of the given number

In [7]:
num = int(input("Enter the Number in Series:"))

n1,n2 = 0,1

if num == 0:
    print("Enter the positive number correctly")
elif num == 1:
    print("The Fibonacci series of 1 :",n1)
else:
    print("The Fibonacci series are")
    
    i = 0
    while i < num:
        print(n1)
        nth = n1 + n2
        n1 = n2
        n2 = nth
        
        i += 1

Enter the Number in Series:10
The Fibonacci series are
0
1
1
2
3
5
8
13
21
34


#### 8.Write a program to find the given number is perfect number
A perfect number is a positive integer that is equal to the sum of its proper divisors (excluding itself). In other words, a perfect number is a number whose divisors (excluding itself) add up to the number itself.

The smallest perfect number is 6, which has divisors 1, 2, and 3, and 1 + 2 + 3 = 6.

Another example is 28: its divisors are 1, 2, 4, 7, and 14, and 1 + 2 + 4 + 7 + 14 = 28.

In [8]:
num = int(input("Enter the Number:"))

a = num
sum = 0

for i in range(1,num): # Avoid num+1 because the divisors of perfect number does not take the number of itself
    if num % i == 0:
        # print(i)
        sum += i
if a == sum:
    print("The Given Number is Perfect")
else:
    print("The Given Number is not Perfect")

Enter the Number:343
The Given Number is not Perfect


#### 9.Write a program to print the perfect number between 1-1000

In [9]:
num = 1000
for i in range(1,num):
    if num % i == 0:
        print(i)

1
2
4
5
8
10
20
25
40
50
100
125
200
250
500


#### 10.Write a program to find the given number is strong number
In Python, a "strong number" is a number in which the sum of the factorials of its digits is equal to the number itself. For example, 145 is a strong number because 1! + 4! + 5! = 1 + 24 + 120 = 145.

In [10]:
num = int(input("Enter the Number:"))

a = num
sum = 0


while num > 0:
    rem = num % 10
    # Find fact for splitted number
    fact = 1
    for i in range(1,rem+1):
        fact = fact * i
        
    sum += fact
    
    # Reduce the number
    num = num // 10
    
if sum == a:
    print("The Given number is Strong Number")
else:
    print("The Given number is not a Strong Number")

Enter the Number:234
The Given number is not a Strong Number


#### 11.Write a program to find the value of one number raised to the power of another

In [11]:
base = int(input("Enter the Base Number :"))
pow = int(input("Enter the Power Number :"))
#res=(base**pow)
#print("Answer :",res)
res = 1
while pow != 0:
    res *= base
    pow-=1
 
print("Answer :",res)

Enter the Base Number :3
Enter the Power Number :2
Answer : 9


#### 12.Write a program to find whether the given number is prime or not

In [12]:
def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, n):
        if n % i == 0:
            return False
    return True

# Test the function
num = int(input("Enter a number: "))
if is_prime(num):
    print(num, "is a prime number")
else:
    print(num, "is not a prime number")

Enter a number: 34
34 is not a prime number


#### 13.Write a program to find the binary number from given decimal number.

In [13]:
decimal = int(input("Enter a decimal number: "))

binary = bin(decimal)

print("Binary representation of", decimal, "is:", binary)

Enter a decimal number: 23
Binary representation of 23 is: 0b10111


In [14]:
def decimal_to_binary(decimal):
    binary = ""
    while decimal > 0:
        binary = str(decimal % 2) + binary
        decimal //= 2
    return binary

# Test the function
decimal = int(input("Enter a decimal number: "))
binary = decimal_to_binary(decimal)
print("Binary representation of", decimal, "is:", binary)

Enter a decimal number: 23
Binary representation of 23 is: 10111


#### 14.The distance between two cities (in km.) is input through the keyboard. Write a program to convert and print this distance in meters, feet, inches and centimeters.

In [15]:
'''
Formulas:
meters = kilometers × 1000
centimeters = meters × 100
inches = centimeters ÷ 2.54
feet = inches ÷ 12
'''
km=float(input("Enter The Kilometer : "));
m=km*1000;
cm=m*100;
i=cm/2.54;
ft=i/12;
print("Kilometers : ",km);
print("Meters  : ",m);
print("Centimeters : ",cm);
print("Inches : ",i);
print("Feet : ",ft);

Enter The Kilometer : 20
Kilometers :  20.0
Meters  :  20000.0
Centimeters :  2000000.0
Inches :  787401.5748031496
Feet :  65616.79790026246


#### 15.Temperature of a city in Fahrenheit degrees is input through the keyboard. Write a program to convert this temperature into Centigrade degrees.( F − 32 ) × 5 / 9 

In [16]:
"""
centigrade = 5/9 * (F-32)

"""
fr=float(input("Enter the Temperature:"))
cent=5.0/9.0*(fr-32)
print("Temparture in Centigrade=",cent)

Enter the Temperature:45
Temparture in Centigrade= 7.222222222222222


#### Prepared By,
Ahamed Basith